### PR TITLE
add ingress classname in ingress.yaml

### PR DESCRIPTION
**Pull request by dmitrygribenchuk**
Add ClassName for ingress controller

### CODE REVIEW

**Comment by mikesindieiev** [comment]
@dmitrygribenchuk we have to add `className` to the [values.yaml](https://github.com/directus-labs/helm-chart/blob/master/charts/directus/values.yaml) as well.

File: charts/directus/templates/ingress.yaml [hunk 1]
```diff
@@ -18,6 +18,7 @@ metadata:
     {{- toYaml . | nindent 4 }}
   {{- end }}
 spec:
+  ingressClassName: {{ .Values.ingress.className }}
```
Comment (mikesindieiev) [inline charts/directus/templates/ingress.yaml:21]:
```suggestion
  {{- if and .Values.ingress.className (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion) }}
  ingressClassName: {{ .Values.ingress.className }}
  {{- end }}
```

**Comment by Hobbit44** [comment]
please could this be added?

**Comment by mikesindieiev** [comment]
Most probably, I will create a fork and maintain helm chart there on my own

**Comment by dmitrygribenchuk** [comment]
> @dmitrygribenchuk we have to add `className` to the [values.yaml](https://github.com/directus-labs/helm-chart/blob/master/charts/directus/values.yaml) as well.

Added.

**Comment by mikesindieiev** [comment]
@dmitrygribenchuk thank you for your effort but it seems like this repo is dead and nobody goes to merge this MR

**Comment by mikesindieiev** [comment]
For those who is struggling with this helm chart but wants to continue use it, you can fine tune helm output with kustomize script by passing `--post-renderer=script.sh` key to the helm command. 
Here is one of the many examples how to do that - https://www.hairizuan.com/altering-outputs-of-helm-installations-with-post-renderer-via-kustomize/

**Comment by Hobbit44** [comment]
@phazonoverload Could @mikesindieiev be given access to contribute to this repo directly if its not being maintained in house?

**Comment by phazonoverload** [comment]
See: https://github.com/directus-labs/helm-chart/issues/22

**Comment by mikesindieiev** [comment]
@dmitrygribenchuk alright, we are half way there
The `helm template` command gives me the following output for the ingress resource:
```yaml
# Source: directus/templates/ingress.yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: directus
  labels:
    helm.sh/chart: directus-0.3.1
    app.kubernetes.io/name: directus
    app.kubernetes.io/instance: directus
    app.kubernetes.io/version: "10"
    app.kubernetes.io/managed-by: Helm
spec:
  ingressClassName:
  rules:
    - host: "chart-example.local"
      http:
        paths:
          - path: /
            backend:
              service:
                name: directus
                port:
                  number: 80
```

I assume we have to fill out `className` key in the `values.yaml` with some value to generate the valid ingress k8s manifest. Let's say `className: "change-me"`

File: charts/directus/values.yaml [hunk 2]
```diff
@@ -42,6 +42,7 @@ service:
 
 ingress:
   enabled: false
+  className: ""
```
Comment (mikesindieiev) [inline charts/directus/values.yaml:45]:
```suggestion
  className: "change-me"
```

**Pull request merged**